.github/workflows/java-ci.yml

```diff
@@ -95,7 +95,7 @@ jobs:
     runs-on: ubuntu-22.04
     strategy:
       matrix:
-        jvm: [11, 17, 21]
```

Comment:
This turns 11 off for our earlier versions, are we able to check with 11 just skipped for 4.0? I don't think its a huge loss if we stop testing 11 but it would be better if we could maintain that

Reply:
Java 11 compatibility is still guarded by `javac`/`scalac` compile options `-release:11`, in other words, jar compiled via Java 17/21 still works with Java 11 runtime.
https://github.com/apache/iceberg/blob/013d09e473faada1a3b21c155964ec4d1254d69d/build.gradle#L189-L192
https://github.com/apache/iceberg/blob/013d09e473faada1a3b21c155964ec4d1254d69d/build.gradle#L238-L246

Reply:
I'll look into this and see if we can still run checks with Java 11, just skipping it specifically for 4.0.